### PR TITLE
`vis` cleanup

### PR DESCRIPTION
**Pull request by gipert**
* type hints and docstring updates
* basic WaveformBrowser test

### CODE REVIEW

**Comment by jasondet** [comment]
Does this overlap with [PR 310](https://github.com/legend-exp/pygama/pull/310)?

Also, should be on main not refactor.

**Comment by jasondet** [comment]
@iguinn also please review this one.

File: src/pygama/vis/waveform_browser.py [hunk 1]
```diff
@@ -17,113 +18,127 @@
...
+                 entry_list: list[int] | list[list[int]] = None,
+                 entry_mask: list[int] | list[list[int]] = None,
+                 dsp_config: str = None,
+                 database: str | dict = None,
+                 aux_values: pandas.DataFrame = None,
+                 lines: str | list[str] = 'waveform',
+                 styles: dict[str, list] | str = None,
+                 legend: str | list[str] = None,
+                 legend_opts: dict = None,
+                 n_drawn: 1 = 1,
```
Comment (iguinn) [inline src/pygama/vis/waveform_browser.py:42]:
```suggestion
                 n_drawn: int = 1,
```

**Comment by iguinn** [comment]
I reviewed the annotations and comment changes to waveform_browser, and aside from the comment it looks good to me.

**Comment by jasondet** [comment]
@gipert can you check why `Build documentation` is failing? I couldn't figure it out from the [Details](https://github.com/legend-exp/pygama/runs/7490788267?check_suite_focus=true).

**Comment by jasondet** [comment]
The build_documentation failure is a https://github.com/sphinx-doc/sphinx/issues/10701. I have turned off this requirement for merging and will merge.

**Pull request merged**